brianlong/www.validators.app
# 181282897 Add pagination to ping thing API

#### What's this PR do?
- Let users paginate results, by default 240 on each page, max 9999 results on each page

#### How should this be manually tested?
- On localhost: create around 1k of results 
```
1000.times do PingThing.create user_id: User.first.id, reported_at: rand(10.hours).seconds.ago, amount: 1, signature: '0d7f418e4d1a3f80dc8a266cd867f766b73d9c80feea36524dfd074068bdef9221e356c192ac6ac71b71404d', response_time: rand(5000..10000), transaction_type: 'transfer', network: 'mainnet', commitment_level: 'processed', application: 'CLI' end 
```
- Check if sorting is okay - check the first and the last record and compare reported_at against view under localhost:3000/ping-thing: ```curl -H "Token: YOURTOKEN" 'http://localhost:3000/api/v1/ping-thing/mainnet' ```
- Confirm that pagination for default limit (240) works as expected: ```curl -H "Token: YOURTOKEN" 'http://localhost:3000/api/v1/ping-thing/mainnet?page=2' ```
- Test various combinations of limit and page, eg.: ```curl -H "Token: YOURTOKEN" 'http://localhost:3000/api/v1/ping-thing/mainnet?limit=10?page=1' ``` ```curl -H "Token: YOURTOKEN" 'http://localhost:3000/api/v1/ping-thing/mainnet?limit=10?page=2' ```


#### What are the relevant tickets?
-[ URL to the PivotalTracker ticket](https://www.pivotaltracker.com/n/projects/2177859/stories/181282897)

#### Task completed checklist:
- [] Is there appropriate test coverage?
- [] Did I take into consideration possible security vulnerabilities?
- [] Are the controllers secure?
- [] Was every important matter documented?

#### Things to watch out for
- [N] Does this PR have any migrations?
- [N] Does this PR add new dependencies?
